hack/swagger/main.go

```diff
@@ -19,64 +19,43 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"runtime/debug"
 	"strings"
 
 	"k8s.io/klog/v2"
 	"k8s.io/kube-openapi/pkg/common"
+	builderutil "k8s.io/kube-openapi/pkg/openapiconv"
 	"k8s.io/kube-openapi/pkg/validation/spec"
 
 	trainer "github.com/kubeflow/trainer/pkg/apis/trainer/v1alpha1"
 )
 
 // Generate Kubeflow Training OpenAPI specification.
 func main() {
-	// Get Kubernetes and JobSet version
-	var k8sVersion string
-	var jobSetVersion string
-
-	info, ok := debug.ReadBuildInfo()
-	if !ok {
-		fmt.Println("Failed to read build info")
-		return
-	}
-
-	for _, dep := range info.Deps {
-		if dep.Path == "k8s.io/api" {
-			k8sVersion = strings.Replace(dep.Version, "v0.", "v1.", -1)
-		} else if dep.Path == "sigs.k8s.io/jobset" {
-			jobSetVersion = dep.Version
-		}
-	}
-	if k8sVersion == "" || jobSetVersion == "" {
-		fmt.Println("OpenAPI spec generation failed. Unable to get Kubernetes and JobSet version")
-		return
-	}
-
-	k8sOpenAPISpec := fmt.Sprintf("https://raw.githubusercontent.com/kubernetes/kubernetes/refs/tags/%s/api/openapi-spec/swagger.json", k8sVersion)
-	// TODO (andreyvelich): Use the release version once this JobSet commit is released: d5c7bce.
-	// jobSetOpenAPISpec := fmt.Sprintf("https://raw.githubusercontent.com/kubernetes-sigs/jobset/refs/tags/%s/hack/python-sdk/swagger.json", jobSetVersion)
-	jobSetOpenAPISpec := "https://raw.githubusercontent.com/kubernetes-sigs/jobset/d5c7bcebe739a4577e30944370c2d7a68321a929/hack/python-sdk/swagger.json"
-
 	var oAPIDefs = map[string]common.OpenAPIDefinition{}
 	defs := spec.Definitions{}
 
 	refCallback := func(name string) spec.Ref {
-		if strings.HasPrefix(name, "k8s.io") {
-			return spec.MustCreateRef(k8sOpenAPISpec + "#/definitions/" + swaggify(name))
-		} else if strings.HasPrefix(name, "sigs.k8s.io/jobset") {
-			return spec.MustCreateRef(jobSetOpenAPISpec + "#/definitions/" + swaggify(name))
-		}
 		return spec.MustCreateRef("#/definitions/" + swaggify(name))
-
 	}
 
 	for k, v := range trainer.GetOpenAPIDefinitions(refCallback) {
 		oAPIDefs[k] = v
 	}
 
 	for defName, val := range oAPIDefs {
-		defs[swaggify(defName)] = val.Schema
+		// Exclude InternalEvent from the OpenAPI spec since it requires runtime.Object dependency.
+		if defName != "k8s.io/apimachinery/pkg/apis/meta/v1.InternalEvent" {
+			// OpenAPI generator incorrectly creates models if enum doesn't have default value.
+			// Therefore, we make the default equal to the first enum value.
+			// Kubernetes OpenAPI spec doesn't have enums: https://github.com/kubernetes/kubernetes/issues/109177
+			for property, schema := range val.Schema.Properties {
+				if schema.Enum != nil && schema.Default == "" {
+					schema.Default = schema.Enum[0]
```

Comment:
```suggestion
					schema.Default = nil
					schema.Nullable = true					
```

Actually, this is not correct. The "" default enum indicates required fields. So, there are no default fields.
Doesn't this generator work fine with nullable=true?

Reply:
Good point, it looks to be working fine: https://github.com/kubeflow/trainer/pull/2490/commits/5094897f0eb0f4a00ff6118406e3d7dc4f92c09c